module2/dao/src/main/java/com/epam/esm/dao/entity/GiftCertificate.java

```diff
@@ -1,29 +1,76 @@
 package com.epam.esm.dao.entity;
 
+import com.epam.esm.dao.audit.GiftCertificateAuditListener;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
-import lombok.Data;
+import lombok.Getter;
 import lombok.NoArgsConstructor;
+import lombok.Setter;
 
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EntityListeners;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.JoinTable;
+import javax.persistence.ManyToMany;
+import javax.persistence.Table;
 import java.math.BigDecimal;
 import java.time.Instant;
 import java.util.List;
 
 /**
  * Represents a Gift certificate.
  */
+@Table(name = "gift_certificate")
+@Entity
 @Builder
-@Data
+@Setter
+@Getter
 @AllArgsConstructor
 @NoArgsConstructor
+@EntityListeners({GiftCertificateAuditListener.class})
 public class GiftCertificate {
 
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
     private Long id;
+    @Column(name = "name", nullable = false, unique = true)
```

Comment:
You can remove @Column here. Hibernate will parse name automatically

Reply:
fixed